hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/AbstractGenericRecordBuilder.java

```diff
@@ -127,6 +128,12 @@ public GenericRecordBuilder setTimestampWithTimezone(@Nonnull String fieldName,
         return write(fieldName, value, FieldKind.TIMESTAMP_WITH_TIMEZONE);
     }
 
+    @Nonnull
+    @Override
+    public GenericRecordBuilder setUuid(@Nonnull String fieldName, @Nullable UUID value) {
```

Comment:
Hi @RiVogel, the changes you did in this PR adds UUID type to the wire level/API of the Compact and Portable serializations, which we decided not to do (because of lack of UUID type in some languages and SQL, and to keep the number of supported types, hence the APIs small, as long as the type can be represented by some other type we support). cc @ihsandemir @srknzl

Reply:
exactly. @RiVogel Did you see the comment by @mdumandag at [here](https://github.com/hazelcast/hazelcast/issues/23698#issuecomment-1471639654)? Is this not a good solution for you? Can you tell us any drawbacks of that approach so that we can understand the need better.

Reply:
Yes, that was no solution for me at all since I'm not able to modify these classes. Since "java.*" is listed in the UNSUPPORTED_PACKAGE_PREFIXES adding UUID support for the compact serializer was the only option.

Reply:
@RiVogel Just a note, you are aware of the overhead of the zero-config Compact serialization, right? It uses reflection to serialize and deserialize data, which incurs considerable overhead

Reply:
I updated the PR. I only change code now within compact serialization.